block-storage/src/test/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorageTest.scala

```diff
@@ -264,4 +266,76 @@ class BlockDagKeyValueStorageTest extends BlockDagStorageTest {
       }
     }
   }
+
+  "recording of new directly finalized block" should "record finalized all non finalized ancestors of LFB" in
```

Comment:
This test is created instead of adjusted FinalizerTest, because what exactly should be finalized having new directly finalised hash - is defined by content of BlockDagStorage.